core/src/main/java/com/linecorp/armeria/internal/server/FileAggregatedMultipart.java

```diff
@@ -61,15 +63,30 @@ public ListMultimap<String, MultipartFile> files() {
         return files;
     }
 
-    public static CompletableFuture<FileAggregatedMultipart> aggregateMultipart(ServiceRequestContext ctx,
-                                                                                HttpRequest req) {
+    public static boolean shouldHandle(BodyPart bodyPart, List<String> parameters) {
+        final String name = bodyPart.name();
+        assert name != null;
+        if (parameters.isEmpty()) {
+            return true;
+        }
+        return parameters.contains(name);
+    }
+
+    public static CompletableFuture<FileAggregatedMultipart> aggregateMultipart(
+            ServiceRequestContext ctx, HttpRequest req) {
+        final Multipart multipart = Multipart.from(req);
+        return aggregateMultipart(ctx, multipart);
+    }
+
+    public static CompletableFuture<FileAggregatedMultipart> aggregateMultipart(
+            ServiceRequestContext ctx, Multipart multipart) {
         final Path destination = ctx.config().multipartUploadsLocation();
-        return Multipart.from(req).collect(bodyPart -> {
+        return Multipart.of(multipart.bodyParts())
+                        .collect(bodyPart -> {
```

Comment:
Question) Can't the `Multipart` just be used as is?

```suggestion
        return multipart.collect(bodyPart -> {
```

Reply:
Fix it!